### ISSUE TITLE
Several Species of Small Furry Animals ___________(next branch; next release)

### ISSUE THREAD

**Issue by tarsius** [magit/magit#1220]
**Usage:** To use Magit's `next` branch you also need Git-mode's `next` branch. You should always update both repositories at the same time. Dash is also required; get it [here](https://github.com/magnars/dash.el). You might also want to remove all customization and then rebuild it after having updated.

---

**Status:** The following things have to be completed before `next` can be merged into `master`:
- [x] #877 Improve cherry-pick and interactive rebase. This is necessary because the Magit specific "rewrite" functionality has already been removed and Magit's cherry-pick support cannot quite replace it yet.
-  [x] #1261 Implement new Ediff support. The old implementation was removed and people would miss it without a substitute.
- Postponed until after `2.1.0` ~~Finish the new popup functionality. This includes #1321~~ and #876 but also a little more.
- [ ] Document important new functionality. Actually properly document everything, including old stuff, see #1096.
- [ ] Fix all the known nasty bugs.

---

**Roadmap toward the next release:**

The next release will be named `2.1.0`. This is inspired by now Emacs is versioned, i.e. there never is a `N.0.M` release.

Currently the snapshots on Melpa and Melpa-Stable have different names but the respective latest snapshots are otherwise identical. I had to make recent Magit versions available on Melpa-Stable, because users were expecting something more recent than `1.2.1`.

Once the above issues have been taken care of Magit will enter feature freeze. I will recreate `maint` from `master`. The snapshots on Magit-Stable will be build by tagging the tip of that branch. But because I have stopped backporting features (see below), I don't expect there to be many such snapshots. Then I will merge `next` into `master` and remove the former.

Melpa (unstable) will continue to build packages from the `master` branch, which now includes the numerous changes previously only in `next`. Currently only a handful users (that I am aware of) are using the `next` branch, so once the majority of Magit users starts using that version, regressions will come to light.

So there then needs to be a several week period to iron out these issues before `2.1.0` can be released. Until that happens, users who prefer suffering the idiosyncrasies of the old version over suffering the less familiar regressions in the new version, can stick to the former by using Melpa-Stable.

Once `2.1.0` is released, the version on Melpa-Stable will always be the latest actual release (whether minor or major, feature or bugfix) and the version on Melpa will once more be a truly bleeding edge snapshot, as it used to be until the the separation of `master` and `next` earlier this year.

---

**No more backporting:** Taking effect immediately I will no longer backport features and fixes to insignificant misfeatures from `next` to `master`. This allows me to focus my efforts on getting `next` ready for the merge into `master`.

**Comment by vermiculus** [comment]
For the record, the referenced roadmap was #1645.
